test/pkg.jl

```diff
@@ -100,6 +100,8 @@ temp_pkg_dir() do
     str = chomp(takebuf_string(iob))
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
+    # Checking to see dependency reading
+    @test Pkg.dependencies("Example") == Dict("julia" => Pkg.Types.VersionSet(v"0.3.0", typemax(VersionNumber)))
```

Comment:
When you say Pkg functions, do you mean the internal ones or the ones which are exposed?

Of the exposed functions, I think only `Pkg.available` and `Pkg.dependents` use only the METADATA files and I think at least `dependents` has unintuitive behavior because of it. For example:

```
julia> Pkg.dependencies("ComputeFramework")
Dict{ByteString,Base.Pkg.Types.VersionSet} with 3 entries:
  "julia" => [0.4.0,∞)
  "Compat" => [0.0.0-,∞)
  "Logging" => [0.0.0-,∞)

julia> Pkg.dependents("Logging")
9-element Array{AbstractString,1}:
 "RobustLeastSquares"
 "JuliaFEM"          
 "JuliaWebAPI"       
 "Mocha"             
 "LoggedDicts"       
 "HTSLIB"            
 "Playground"        
 "EEG"               
 "EcologicalNetwork" 
```

Where ComputeFramework doesn't show up as a dependent of Logging.